### PR TITLE
LPS-189866 UpgradeJavaSearchVocabulariesMethodCheck

### PR DESCRIPTION
**Pull request by TamyrisTorres**
Ticket: https://liferay.atlassian.net/browse/LPS-189866

Hi Kevin!

For this check, we need to add new parameters to the method. 
Still, I did create a method to compare the types of parameters.

Please, Can you review this code?

### CODE REVIEW

**Comment by TamyrisTorres** [comment]
ci:test:sf

**Comment by TamyrisTorres** [comment]
ci:test:relevant

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java [hunk 1]
```diff
@@ -80,6 +80,23 @@ public int getWeight() {
...
+			if ((variableTypeName == null) ||
+				!parameterTypes[i].equals(variableTypeName)) {
+
+				return false;
+			}
+		}
+
+		return true;
+	}
+
```
Comment (kevhlee) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java:99]: Hi @TamyrisTorres, I know we discussed this earlier.

I have some concerns about this particular automation/rule and the `hasParameterTypes` method. Checking the types of the arguments in method call is very difficult and error-prone when we are dealing with strings, especially when we consider literal values and method calls within method calls. Even with a dedicated Java parser, it is still a little tricky based on the SF framework's current capabilities.

I question whether this should be automated. At best, only the check should only handle the case when all the method arguments are variables. If the arguments are not all variables, then we ignore it. Let me know if that makes sense.

**Review by kevhlee — COMMENTED** [review]
Hi @TamyrisTorres, I know we discussed this earlier.

I have some concerns about this particular automation/rule and the `hasParameterTypes` method. Checking the types of the arguments in method call is very difficult and error-prone when we are dealing with strings, especially when we consider literal values and method calls within method calls. Even with a dedicated Java parser, it is still a little tricky based on the SF framework's current capabilities.

I question whether this should be automated. At best, only the check should only handle the case when all the method arguments are variables. If the arguments are not all variables, then we ignore it. Let me know if that makes sense.

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java [hunk 2]
```diff
@@ -80,6 +80,23 @@ public int getWeight() {
 		return _weight;
 	}
 
+	public boolean hasParameterTypes(
+		String content, List<String> parameterList, String[] parameterTypes) {
```
Comment (kevhlee) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java:84]: Can we convert `parameterList` to an array instead? Mixing a list and an array like this is a bit odd and `parameterList.get(i)` could be slow if the list is something like a `LinkedList`.

**Review by kevhlee — CHANGES_REQUESTED** [review]

**Comment by nicolas-sss** [comment]
> Hi @TamyrisTorres, I know we discussed this earlier.
> 
> I have some concerns about this particular automation/rule and the `hasParameterTypes` method. Checking the types of the arguments in method call is very difficult and error-prone when we are dealing with strings, especially when we consider literal values and method calls within method calls. Even with a dedicated Java parser, it is still a little tricky based on the SF framework's current capabilities.
> 
> I question whether this should be automated. At best, the check should only handle the case when all the method arguments are variables. If the arguments are not all variables, then we ignore it. Let me know if that makes sense.

Hi @kevhlee, Since Tamyris has gone on vacation, I will be responsible for those PR's of hers in your queue. That said I agree with you that these types of verifications are tricky dealing with strings. 
We will have three possibilities types for parameters:

1. Variables
2. literal types, like `"48484"` as a long parameter
3. Method returns, like `"someVariable.getSomething()"` This is the impossible case because we have no way of knowing the return of this method of this outer class.

So, knowing this I thought this should be the approach:

1- easy with getVariableType
2- Try somehow conversions to literal types, if none type is true is the last case automatically.
3- As it is not a possible case, send a message warning that the specific method could not be automated.

What do you think? Do you still think that only the case of all parameters being variables should be automated? What do we do with the other two? Send a message?

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java (hunk 2 above)
Comment (nicolas-sss) [reply modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java:84]: Yes, we can!

**Comment by kevhlee** [comment]
Hey @nicolas-sss, it is fine to automate the variable case. However, for any other case, even trying to convert the arguments into literal types is tricky and slow (we would literally need to try every literal type conversion). I think your suggestion of sending a warning message is better.

**Comment by nicolas-sss** [comment]
Resent at https://github.com/kevhlee/liferay-portal/pull/122

**Comment by nicolas-sss** [comment]
@kevhlee please close this one, I have sent you a new one about this. Thanks.

**Pull request closed without merging**